### ISSUE TITLE
Output to stdout not working with multiple output formatters

### ISSUE THREAD

**Issue by max-moritz** [stoplightio/spectral#2042]
**Describe the bug**

spectral lint fails with error message `The number of outputs must match the number of formats` when on the command line there are two formatters specified but only one output file

**To Reproduce**
1. Given an arbitrary OpenAPI document `my-api.yaml` and any ruleset `.spectral.yaml`
2. Run this CLI command: `spectral lint --format stylish --format junit --output.junit junit-results.xml my-api.yaml`
3. See error  `[usage message omitted] The number of outputs must match the number of formats`

**Expected behavior**

junit formatted output is written to junit-results.xml while stylish formatted output is written to stdout.

Expecting this behavior since the usage message says
` -o, --output                   where to output results, can be a single file name, multiple "output.<format>" or missing to print to stdout`

**Environment**
 - spectral version: 6.2.0
 - OS: Microsoft Windows 10 Enterprise Version 10.0.19042 Build 19042

**Additional context**
No error when formatter count matches output file count, e.g. with
`spectral lint --format stylish --format junit --output.stylish stylish-results --output.junit junit-results.xml my-api.yml`

**Comment by fgreinacher** [comment]
@P0lip Something seems to be wrong in #2037, weird that there is a [test case for nearly the same command line](https://github.com/stoplightio/spectral/pull/2037/files#diff-5c1bab693e4315f93ddc3bc536502fe23b536bbed6214ca5a093571db6389f24R177) 🤷

**Comment by P0lip** [comment]
Going to look into it in a few mins

**Comment by P0lip** [comment]
~I wonder whether "middleware" isn't executed after "check" for some reason 🤔 
Seems like so.~

nah, it's my fault. The test swallowed the error 🤦‍♂️

**Comment by fgreinacher** [comment]
As a workaround you can pass `--output.stylish "<stdout>"`.

**Comment by leubedane** [comment]
For me this issue is not fixed. It is still needed to use the workaraound. So can you reopen this issue? Or is the workaround the preffered solution?

**Comment by P0lip** [comment]
@leubedane could you toss in the command you're using?

**Comment by menaheme** [comment]
typing in the command:
`spectral lint  --ruleset ./my-ruleset.json --format stylish --format html --output.stylish stylish-results --output.html html-results.html my-api.yml`
i get :
`multiple outputs must be provided when more than a single format is specified`

**Comment by jeremycohensolal-paylead** [comment]
This syntax works for me:

```bash
spectral lint -D -F warn \
        -f junit -o.junit "junit-report.xml" \
        -f stylish -o.stylish "<stdout>" \
        --ruleset "ruleset.yaml" "openapi.yaml"
```

NOTE: the `<stdout>` add a print of the stylish format to stdout (usefull in my gitlab job)
